ballerina/tests/test.bal

```diff
@@ -0,0 +1,243 @@
+    }
+}
+
+// Test function for retrieving a batch of quotes 
+@test:Config{}
+function testGetBatchOfQuotes() returns error? {
+
+    SimplePublicObjectId ob0 = {
+        id: testQuoteId 
+    };
+
+    BatchReadInputSimplePublicObjectId payload = {
+        properties: [],
+        propertiesWithHistory: [], 
+        inputs: [ob0]
+    };
+
+    BatchResponseSimplePublicObject|BatchResponseSimplePublicObjectWithErrors response = check hubspotClient->/batch/read.post(payload);
+
+    // Validate essential fields
+    test:assertTrue(response.results.length() == payload.inputs.length(), msg = string`Only ${response.results.length()} IDs found.`);
+}
+
+
+// Archive a quote by ID
+@test:Config{}
+function testArchiveOneQuote() returns error?{
+
+    http:Response|error response = check hubspotClient->/["0"].delete(); 
+
+    // Validate the response
+    if(response is http:Response){
+        test:assertTrue(response.statusCode == 204);
+    }else{
+        test:assertFail("Deletion failed."); 
+    }
+}
+
+// Archive batch of quotes by ID
+@test:Config{}
+function testArchiveBatchOfQuoteById() returns error?{
+
+    SimplePublicObjectId id0 = {id:"0"};
+
+    BatchInputSimplePublicObjectId payload = {
+        inputs:[
+            id0 
+        ]
+    };
+
+    http:Response|error response = check hubspotClient->/batch/archive.post(payload); 
+
+    // Validate the response
+    if(response is http:Response){
+        test:assertTrue(response.statusCode == 204);
+    }else{
+        test:assertFail("Deletion failed."); 
+    }
+}
+
+
+// Test function for updating a quote
+@test:Config{}
+function testUpdateOneQuote() returns error? {
+    SimplePublicObjectInput payload = {
+        properties: {
+            "hs_title": "Test Quote Modified",
+            "hs_expiration_date": "2025-03-31" 
+        }
+    };
+
+    // Call the Quotes API to update the quote
+    SimplePublicObject|error response = check hubspotClient->/[testQuoteId].patch(payload);
+
+    // Validate the response
+    if(response is SimplePublicObject){
+        test:assertTrue(response.id != "", 
+        msg = "Quote in response does not match the expected quote."); 
+    }else {
+        io:println(response); 
+    }
+}
+
+// Test function for updating a batch of quotes
+@test:Config{}
+function testUpdateBatchOfQuotes() returns error? {
+
+    SimplePublicObjectBatchInput ob3 = {
+        id: testQuoteId,
+        properties: {
+            "hs_title": "Test Quote 3", 
+            "hs_expiration_date": "2025-04-30"
+        }
+    };
+
+    BatchInputSimplePublicObjectBatchInput payload = {
+        inputs: [ob3]
+    };
+
+    // Call the Quotes API to create a new quote
+    BatchResponseSimplePublicObject|BatchResponseSimplePublicObjectWithErrors|error response = hubspotClient->/batch/update.post(payload);
+
+    // Validate the response
+    if(response is BatchResponseSimplePublicObject){
```

Comment:
```suggestion
    if response is BatchResponseSimplePublicObject {
```
Braces in if statements are optional in Ballerina. Let's review and remove them in other places where applicable.